### DISCUSSION TITLE
Connecting to a Codespace from within a PWA

### DISCUSSION THREAD

**Discussion by mpriscella** [community/community#10967] [Codespaces]
When using [VSCode as a PWA](https://code.visualstudio.com/updates/v1_64#_pwa-and-offline-support), attempting to connect to a Codespace through the "Remote Explorer" tab takes you out of the PWA and opens the Codespace in your default browser. Is it possible to connect from within the same PWA instance and not in a new window?

This is probably not a common use case on desktop operating systems, but this functionality would be really useful on iPads or other tablets.

**Comment by m0xsec** [comment]
bumping this, I ran into this while using the PWA on a tablet.

**Comment by Jaber-Saed** [comment]
As of my last knowledge update in September 2021, Visual Studio Code's support for Progressive Web Apps (PWAs) was still evolving, and the specific behavior you're describing might have been introduced or changed after that point. However, I can provide some general insights based on the state of technology at that time.

VSCode as a PWA does have limitations compared to the full desktop application, primarily due to the sandboxed environment of PWAs and the limitations of web technologies. While VSCode's web version (VS Code Online) does offer many features, there might still be gaps between the PWA version and the full desktop application.

As for the behavior you're describing—connecting to a Codespace within the PWA instance instead of opening it in a new browser window—this could depend on how the integration between VSCode PWA and Codespaces has been implemented. If the integration is designed to launch Codespaces in a separate browser window, it might be challenging to change this behavior within the PWA's sandboxed environment.

If you're using the PWA version of VSCode and encountering this issue, here are a few steps you can consider:

1. **Check Updates:** Ensure that you're using the latest version of the VSCode PWA. Developers are continually working on improving the PWA experience, and updates might include changes to how certain features behave.

2. **Documentation and Feedback:** Check the official documentation for VSCode's PWA version or their forums/community for any discussions or updates related to connecting to Codespaces. If you don't find the feature you're looking for, consider providing feedback to the VSCode team about this use case and its importance, especially for tablet users.

3. **Alternative Tools:** If connecting to Codespaces from within the PWA instance isn't supported, you might want to explore other tools, apps, or workflows that allow better integration between Codespaces and tablet devices.

4. **Future Updates:** Given that the technology landscape evolves rapidly, it's possible that the behavior you're describing could change in future updates. Stay tuned to official announcements and updates related to VSCode's PWA version.

Remember that my information is based on the state of technology as of September 2021, and there might have been developments or changes since then. If you're looking for the most up-to-date and accurate information, I recommend checking official documentation, forums, and the product's website or blog for announcements and updates.
